### PR TITLE
UI dynamic collectable

### LINKED ISSUE #79: #4 Dynamically create collected mountain items
Create so that the collected mountains dynamically fills a list of the collected mountains and that the mountains are clickable and able to send information to the next activity when clicked

### PR DESCRIPTION
**Pull request by aolsson711**
Created so the collected item list in collection is dynamically loaded, where each button starts an activity with the correct information. 

closes #79

### CODE REVIEW

File: app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/CollectionActivityTest.java [hunk 1]
```diff
@@ -58,12 +65,44 @@ public void TestToolbarBackButton(){
...
 
-    /* Test that pressing the collected item the view changes to MountainActivity */
+    /* Test that all elements in list view are at correct place and contains correct data */
     @Test
-    public void TestCollectedItem(){
-        ViewInteraction button = Espresso.onView(withId(R.id.collected));
-        button.perform(ViewActions.click());
-        // Catch intent
-        intended(IntentMatchers.hasComponent(MountainActivity.class.getName()));
+    public void TestContentOfListView(){
```
Comment (giommok) [inline app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/CollectionActivityTest.java:70]: When we will display the actual data this test will have to be reworked, right? Maybe we should add a TODO comment? Also, what will we display when no internet connection is available? Maybe we can test this case as of now.

**Review by giommok — CHANGES_REQUESTED** [review]

**Review by giommok — APPROVED** [review]
Looks good!

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/CollectionListAdapter.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * List adapter for collected items to set correct text for each collected item.
+ */
+public class CollectionListAdapter extends ArrayAdapter<CollectedItem> {
+
+    private final int resourceLayout;
+    private final Context mContext;
+
+    public CollectionListAdapter(Context context, int resource, List<CollectedItem> items)
```
Comment (magangloff) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/CollectionListAdapter.java:21]: Add javadoc for constructor, curly braces are not consistent

**Review by magangloff — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/collected_item.xml [hunk 3]
```diff
@@ -1,23 +1,15 @@
...
-        android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:background="@drawable/button_light_green">
-    </Button>
+    android:layout_height="wrap_content"
+    android:layout_gravity="center"
+    android:background="@drawable/button_light_green">
 
+    <!-- Container -->
     <RelativeLayout
```
[the remark this answers is not in the document]
Comment (aolsson711) [reply app/src/main/res/layout/collected_item.xml:10]: Needed for tests

File: app/src/main/res/layout/ranking_item.xml [hunk 4]
```diff
@@ -1,5 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!-- TODO: Remake to be able to create dynamically -->
 <RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
[the remark this answers is not in the document]
Comment (aolsson711) [reply app/src/main/res/layout/ranking_item.xml:2]: Needed for tests

**Review by magangloff — APPROVED** [review]
All good

**Pull request merged**